cdap-proto/src/main/java/co/cask/cdap/proto/AdapterDetail.java

```diff
@@ -16,24 +16,28 @@
 
 package co.cask.cdap.proto;
 
-import com.google.common.base.Objects;
+import co.cask.cdap.api.schedule.ScheduleSpecification;
 import com.google.gson.JsonElement;
 
 /**
- * Adapter details that come back from get calls.
- * TODO: finalize what should be present here
+ * Represents an adapter returned for /adapters/{adapter-id}.
  */
-public final class AdapterDetail {
+public class AdapterDetail {
   private final String name;
   private final String description;
   private final String template;
+  private final Id.Program program;
```

Comment:
May we should have a flat representation for program id in proto for all REST endpoints. E.g. Just

```java
class ProgramId {
  String namespace;
  String application;
  String type;
  String id;
}
```